core/client_test.go

```diff
@@ -110,6 +110,13 @@ func TestRemoteClient_RetryDial(t *testing.T) {
 	remote := StartMockNode(CreateKvStore(defaultRetainBlocks))
 	protocol, ip := getRemoteEndpoint(remote)
 
+	// wait for remote client to run
+	for {
+		if remote.IsRunning() {
+			break
+		}
+	}
+
```

Comment:
I'll just loop max 5x to see if client's running, pls check now. I think it needs a test.

Reply:
It's unclear now how this behaves. I would assume that one iteration just takes a few ns to ms. It will probably exhaust these 5 loops immediately and not really wait for the client to be up?

Reply:
Come to think of it, this test adds 3 seconds to our total test time (since retry is every second anyway). Maybe I'll just remove the test for this.